strawberry/schema/schema_converter.py

```diff
@@ -336,6 +336,26 @@ def _check_permissions(source: Any, info: Info, kwargs: Dict[str, Any]):
                     message = getattr(permission, "message", None)
                     raise PermissionError(message)
 
+        async def _check_permissions_async(
+            source: Any, info: Info, kwargs: Dict[str, Any]
+        ):
+            for permission_class in field.permission_classes:
+                permission = permission_class()
+                has_permission: bool
+
+                if inspect.iscoroutinefunction(permission.has_permission):
+                    has_permission = await permission.has_permission(  # type: ignore
+                        source, info, **kwargs
+                    )
+                else:
+                    has_permission = permission.has_permission(  # type: ignore
+                        source, info, **kwargs
+                    )
```

Comment:
The `await_me_maybe` function would make this code a bit cleaner (https://simonwillison.net/2020/Sep/2/await-me-maybe/):

```suggestion
                
                has_permission = await await_me_maybe(
                	permission.has_permission(source, info, **kwargs)
                )
```